python/ray/data/datasource/parquet_datasource.py

```diff
@@ -36,38 +37,30 @@
 FILE_READING_RETRY = 8
 
 
-def _register_parquet_file_fragment_serialization():
-    from pyarrow.dataset import ParquetFileFragment
+# TODO(ekl) this is a workaround for a pyarrow serialization bug, where serializing a
+# raw pyarrow file fragment causes S3 network calls.
```

Comment:
Won't this no longer be an issue since we're no longer closing over `self`, which contained the full `ParquetDataset` object? It seems like that was the real source of the issue with the failing nightly test.

Reply:
You still get RPCs serializing the fragments, which are also passed separately independently of self.

Reply:
I'm not saying that we can get rid of all serialization hacks for file fragments, I'm asking if the old method (registering a serialization handler) actually was working as intended for the file fragments that are passed as an argument and that closing over `self` was the issue, which would bypass either of these serialization hacks.

In other words, I'm wondering if we revert back to the old serialization hack of registering a serialization handler (i.e. reverting this commit), whether the nightly test would pass.

Reply:
Possibly, but I'd rather handle things explicitly the same way everywhere rather than debug registration issues. It's unclear what issues would remain with reverting that commit.

Reply:
IMO this class is easier to understand than using registration. Registering and then unregistering right after the code block seems not intuitive to me (the registry may be expected to last the lifecycle of process).

The most confusing part is indeed the RPCs inside these ser/deser. Just to be clear about this: does this make RPCs both in serialization and deserialization? (deserialization probably does because we do retry to fix HDFS issue before).

Reply:
>Registering and then unregistering right after the code block seems not intuitive to me (the registry may be expected to last the lifecycle of process).

Yeah we shouldn't be doing that for the registration pattern, that's not what we do for the JSON read options, for example.

>The most confusing part is indeed the RPCs inside these ser/deser. Just to be clear about this: does this make RPCs both in serialization and deserialization? (deserialization probably does because we do retry to fix HDFS issue before).

The network requests only happen on deserialization AFAIK.